src/main/java/org/testng/reporters/SuiteHTMLReporter.java

```diff
@@ -346,101 +347,110 @@ private String dumpGroups(String[] groups) {
+	String configurationGroupsMethod = isGroupsConfiguration ? (bg ? BEFORE : AFTER) + methodName : SP;
+	String configurationSuiteMethod = isSuiteConfiguration ? (bs ? BEFORE : AFTER) + methodName : SP;
+	String setUpOrTearDownMethod = isSetupOrTearDown ? (setUp ? BEFORE : AFTER) + methodName : SP;
+	String testMethod = tm.isTest() ? methodName : SP;
+
+	StringBuffer instances = new StringBuffer();
+	for (long o : tm.getInstanceHashCodes()) {
+	  instances.append(o).append(" ");
+	}
+
+	if (startDate == -1) {
+	  startDate = iim.getDate();
+	}
+	String date = format.format(iim.getDate());
+	bw.append("<tr bgcolor=\"" + createColor(tm) + "\">")
+	  .append("  <td>").append(date).append("</td> ")
+	  .append("  <td>").append(Long.toString(iim.getDate() - startDate)).append("</td> ")
+	  .append(td(configurationSuiteMethod))
+	  .append(td(configurationTestMethod))
+	  .append(td(configurationClassMethod))
+	  .append(td(configurationGroupsMethod))
+	  .append(td(setUpOrTearDownMethod))
+	  .append(td(testMethod))
+	  .append("  <td>").append(tm.getId()).append("</td> ")
+	  .append("  <td>").append(instances).append("</td> ")
+	  .append("</tr>\n")
+	  ;
       }
-      String methodName = tm.toString();
-      boolean bc = tm.isBeforeClassConfiguration();
-      boolean ac = tm.isAfterClassConfiguration();
-      boolean bt = tm.isBeforeTestConfiguration();
-      boolean at = tm.isAfterTestConfiguration();
-      boolean bs = tm.isBeforeSuiteConfiguration();
-      boolean as = tm.isAfterSuiteConfiguration();
-      boolean bg = tm.isBeforeGroupsConfiguration();
-      boolean ag = tm.isAfterGroupsConfiguration();
-      boolean setUp = tm.isBeforeMethodConfiguration();
-      boolean tearDown = tm.isAfterMethodConfiguration();
-      boolean isClassConfiguration = bc || ac;
-      boolean isGroupsConfiguration = bg || ag;
-      boolean isTestConfiguration = bt || at;
-      boolean isSuiteConfiguration = bs || as;
-      boolean isSetupOrTearDown = setUp || tearDown;
-      String configurationClassMethod = isClassConfiguration ? (bc ? BEFORE : AFTER) + methodName : SP;
-      String configurationTestMethod = isTestConfiguration ? (bt ? BEFORE : AFTER) + methodName : SP;
-      String configurationGroupsMethod = isGroupsConfiguration ? (bg ? BEFORE : AFTER) + methodName : SP;
-      String configurationSuiteMethod = isSuiteConfiguration ? (bs ? BEFORE : AFTER) + methodName : SP;
-      String setUpOrTearDownMethod = isSetupOrTearDown ? (setUp ? BEFORE : AFTER) + methodName : SP;
-      String testMethod = tm.isTest() ? methodName : SP;
-
-      StringBuffer instances = new StringBuffer();
-      for (long o : tm.getInstanceHashCodes()) {
-        instances.append(o).append(" ");
+      bw.append("</table>\n");
+    } catch (IOException e) {
+      Utils.log("[SuiteHTMLReporter]", 1, "Error writing to " + outputFileName + ": " + e.getMessage());
+    } finally {
+      try {
+        if (bw != null) {
+          bw.close();
```

Comment:
Thanks for the info, I just copied this from Utils.java. Anyway, done in e353be223c97f152cb3eb1b6df65200eeed7021f (which I will rebase into a9c69ddb24b95d680bdaab61c6c614bf58d73cb5).